src/main/api/publication.ts

```diff
@@ -179,40 +178,22 @@ export class PublicationApi implements IPublicationApi {
                 this.dispatchToastRequest(ToastType.DownloadFailed, `[${error}]`);
                 throw new Error(`importOpdsPublication ${error}`);
             }
-            titleView = publication.title;
             this.sendDownloadSuccess(url);
             returnView = this.publicationViewConverter.convertDocumentToView(publication);
         }
-
-        // dispatch notification to user with redux
-        this.dispatchToastRequest(ToastType.DownloadComplete,
-            this.translator.translate("message.download.success", { title: titleView }));
         return returnView;
     }
 
-    public async import(paths: string[]): Promise<PublicationView[]> {
-        // returns all publications linked to this import
-        const newDocs = [];
-
-        for (const path of paths) {
-            try {
-                const newDoc = await this.catalogService.importFile(path);
-                if (newDoc) {
-                    newDocs.push(newDoc);
-                }
-            } catch (error) {
-                debug(`Import file - FAIL : ${path}`, error);
-                this.dispatchToastRequest(ToastType.DownloadFailed,
-                    this.translator.translate("message.import.fail", {path}));
-            }
+    public async import(filePathArray: string | string[]): Promise<PublicationView[]> {
+        if (!isArray(filePathArray)) {
+            filePathArray = [filePathArray];
         }
-
-        return newDocs.map((doc) => {
-            const publication = this.publicationViewConverter.convertDocumentToView(doc);
-            this.dispatchToastRequest(ToastType.DownloadComplete,
-                this.translator.translate("message.import.success", { title: publication.title }));
-            return publication;
-        });
+        // returns all publications linked to this import
+        const pubsRawPromise = filePathArray.map((filePath) => this.catalogService.importFile(filePath));
+        const pubsRaw = await Promise.all(pubsRawPromise);
```

Comment:
As soon as one import fails, the entire multi-file import fails.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all

Should we allow individual imports to succeed, even if one of them fails?

Reply:
I missed this.. I propose this fct : https://github.com/readium/readium-desktop/pull/793/commits/dcc68ce955bf2f5cab375784a5ed2667de908fd4

Reply:
Yes, until this:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled
:)